src/actions/product-db-actions.ts

```diff
@@ -0,0 +1,91 @@
+import {Product} from '../db-schemas/product-schema';
+import {ProductDocument} from '../db-schemas/product-document';
+import {logger} from '../logger/logger';
+
+export const retrieveAllProducts = async () => {
+    const products = await Product.find();
+    return products.map((product) => product.toClient());
+};
+
+export const retrieveSpecificProduct = async (id: string) => {
+    const product = await Product.findById(id);
+    if (!product)
+        return null;
+    return product.toClient();
+};
+
+export const createProduct = async (productJson: object) => {
```

Comment:
isn't the type of `productJson`  `ProductDocument`?

Reply:
fixed